zio-json/jvm/src/main/scala/zio/json/JsonDecoderPlatformSpecific.scala

```diff
@@ -30,10 +30,10 @@ trait JsonDecoderPlatformSpecific[A] { self: JsonDecoder[A] =>
   final def decodeJsonStreamInput[R](
     stream: ZStream[R, Throwable, Byte],
     charset: Charset = StandardCharsets.UTF_8
-  ): ZIO[R, Throwable, A] =
+  ): ZIO[R with Scope, Throwable, A] =
     stream.toInputStream
-      .flatMap(is => ZManaged.fromAutoCloseable(UIO(new java.io.InputStreamReader(is, charset))))
-      .use(readAll)
+      .flatMap(is => ZIO.fromAutoCloseable(ZIO.succeed(new java.io.InputStreamReader(is, charset))))
```

Comment:
Anytime you use `.use` in the old ZManaged API, it means you need to use `ZIO.scoped { ... }` in the new API, to close the open resources. Right now, they are left open when the function returns.

Reply:
Would we just wrap the latter part?  e.g.:

```scala
    stream.toInputStream
      .flatMap(is =>
        ZIO.scoped {
          ZIO
            .fromAutoCloseable(ZIO.succeed(new java.io.InputStreamReader(is, charset)))
            .flatMap(readAll)
        }
```